lib/bibliothecary.rb

```diff
@@ -10,9 +10,32 @@
 end
 
 module Bibliothecary
-  def self.analyse(path)
+  def self.analyse(path, error_if_no_parser=false)
     info_list = load_file_info_list(path)
-    package_managers.map{|pm| pm.analyse_file_info(info_list) }.flatten.compact
+    # Each package manager needs to see the entire list so it can
+    # associate related manifests and lockfiles for example.
+    analyses = package_managers.map{|pm| pm.analyse_file_info(info_list) }.flatten.compact
+
+    if error_if_no_parser
```

Comment:
Something about this feels kludgey to me and I'm trying to decide why.

I'm wondering if it's cleaner to think about return a tuple or a hash where you get `analyzed_files` and `ignored_files` or similar. Callers could then choose to not worry about `ignored_files` but there's not this different case that you have to pass in (and also, you then don't keep track of what the analysis hash looks like over time in two places)

Reply:
The way our dependencyci code is written we'd have to build the analysis hash over there if we don't do it here - remember other kinds of parse error are represented this way too. I can definitely clean up that cut-and-paste here somehow though.

I think what makes the method ugly is that we pass all files to all package managers, instead of identifying the package manager for files and then passing the files only to the right ones. I can kinda refactor it up to illustrate.